packages/ethereum/contracts/script/SingleAction.s.sol

```diff
@@ -483,4 +484,52 @@ contract SingleActionFromPrivateKeyScript is Test, EnvironmentConfig {
     }
     vm.stopBroadcast();
   }
+
+  /**
+   * ported from HoprStakeBase.sol
+   * @dev if the given `tokenURI` end with `/substring`
+   * @param tokenURI string URI of the HoprBoost NFT. E.g. "https://stake.hoprnet.org/PuzzleHunt_v2/Bronze - Week 5"
+   * @param substring string of the `boostRank` or `boostType/boostRank`. E.g. "Bronze - Week 5", "PuzzleHunt_v2/Bronze - Week 5"
+   */
+  function _hasSubstring(string memory tokenURI, string memory substring) internal pure returns (bool) {
```

Comment:
Function name is misleading, since you only check for a suffix.

Also would it make sense to use an alternative implementation? E.g. https://github.com/Arachnid/solidity-stringutils/blob/master/src/strings.sol#L417

Reply:
Precisely speaking, this function only check for `/suffix`. The alternative implementation is versatile for more string manipulation. The reason that `_hasSubstring` is ported here is simply because the same implementation was used in `HoprStakeBase` contract. It can be updated to use strings library without any problem.